website/src/components/SearchPage/fields/AutoCompleteField.spec.tsx

```diff
@@ -295,4 +294,256 @@ describe('AutoCompleteField', () => {
+                    multiSelect={true}
+                    fieldValues={['Option 1', 'Option 2']}
+                />,
+            );
+
+            expect(screen.getByText('Option 1')).toBeInTheDocument();
+            expect(screen.getByText('Option 2')).toBeInTheDocument();
+
+            const removeButtons = screen.getAllByLabelText(/Remove/);
+            expect(removeButtons).toHaveLength(2);
+        });
+
+        it('removes individual selected values when badge remove button is clicked', async () => {
+            mockUseAggregated.mockReturnValue({
+                data: {
+                    data: [
+                        { testField: 'Option 1', count: 10 },
+                        { testField: 'Option 2', count: 20 },
+                    ],
+                },
+                isLoading: false,
+                error: null,
+                mutate: vi.fn(),
+            });
+            render(
+                <AutoCompleteField
+                    field={field}
+                    optionsProvider={{
+                        type: 'generic',
+                        lapisUrl,
+                        lapisSearchParameters,
+                        fieldName: field.name,
+                    }}
+                    setSomeFieldValues={setSomeFieldValues}
+                    multiSelect={true}
+                    fieldValues={['Option 1', 'Option 2']}
+                />,
+            );
+
+            const removeButton = screen.getByLabelText('Remove Option 1');
+            await userEvent.click(removeButton);
+
+            expect(setSomeFieldValues).toHaveBeenCalledWith(['testField', ['Option 2']]);
+        });
+
+        it('clears all selections when clear button is clicked', async () => {
+            mockUseAggregated.mockReturnValue({
+                data: {
+                    data: [
+                        { testField: 'Option 1', count: 10 },
+                        { testField: 'Option 2', count: 20 },
+                    ],
+                },
+                isLoading: false,
+                error: null,
+                mutate: vi.fn(),
+            });
+            render(
+                <AutoCompleteField
+                    field={field}
+                    optionsProvider={{
+                        type: 'generic',
+                        lapisUrl,
+                        lapisSearchParameters,
+                        fieldName: field.name,
+                    }}
+                    setSomeFieldValues={setSomeFieldValues}
+                    multiSelect={true}
+                    fieldValues={['Option 1', 'Option 2']}
+                />,
+            );
+
+            const clearButton = screen.getByLabelText('Clear Test Field');
+            await userEvent.click(clearButton);
+
+            expect(setSomeFieldValues).toHaveBeenCalledWith(['testField', '']);
+        });
+
+        it('deselects an option when clicking on already selected option', async () => {
```

Comment:
From the name of the test I would have assumed that it clicks in the autocomplete options dropdown. But instead it clicks on the `x` in the badge, doesn't it?